src/pages/scheduler/components/ShareTimetableButton.tsx

```diff
@@ -0,0 +1,185 @@
+        </Text>
+      )}
+    </Wrap>
+  );
+};
+
+const SocialMediaButtons = (props: { slug: string }) => {
+  return (
+    <HStack justify="center">
+      <EmailShareButton children={<EmailIcon size={50} round={true} />} url={props.slug} />
+      <FacebookShareButton children={<FacebookIcon size={50} round={true} />} url={props.slug} />
+      <TelegramShareButton children={<TelegramIcon size={50} round={true} />} url={props.slug} />
+      <WhatsappShareButton children={<WhatsappIcon size={50} round={true} />} url={props.slug} />
+      <TwitterShareButton children={<TwitterIcon size={50} round={true} />} url={props.slug} />
+    </HStack>
+  );
+};
+
+const CopyLinkUrl = (props: { isSmallScreen: boolean; slug: string }) => {
+  const toast = useToast();
+  return (
+    <HStack justify="space-between" padding="5px" borderWidth="1px" borderColor="gray.300">
+      <HStack justify="center" flexGrow={4}>
+        {props.isSmallScreen ? <Icon boxSize="1.25em" as={HiLink} /> : undefined}
+        <Input id="timetable_slug" value={props.slug} variant="filled" isReadOnly={true} />
+      </HStack>
+      <Button
+        size={props.isSmallScreen ? 'sm' : 'md'}
+        colorScheme="blue"
+        leftIcon={<CopyIcon />}
+        onClick={() => {
+          var el = document.getElementById('timetable_slug') as HTMLInputElement;
+          el.focus();
+          el.select();
+          document.execCommand('copy');
+          toast({
+            title: 'Copied.',
+            status: 'success',
+            duration: 3000,
+          });
+        }}
+      >
+        Copy
+      </Button>
+    </HStack>
+  );
+};
+
+const ShareTimetableContent = (props: { courses: Array<SavedCourse>; isSmallScreen: boolean; term: string }) => {
+  const slug = 'https://courseup.ca/s/9w845yetg9d8gh938wrhsde9';
+
+  return (
+    <VStack align="left" spacing="15px">
+      <InformationText isSmallScreen={props.isSmallScreen} term={props.term} />
+      <Heading size="sm"> What you are sharing </Heading>
+      <SelectedCoursesTable courses={props.courses} term={props.term} />
+      <Heading size="sm"> Share this link via </Heading>
+      <SocialMediaButtons slug={slug} />
+      <Heading size="sm"> Or copy link </Heading>
+      <CopyLinkUrl isSmallScreen={props.isSmallScreen} slug={slug} />
+    </VStack>
+  );
+};
+
+export default function ShareTimetableButton() {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const { term } = useParams();
+
+  const [isSmallScreen] = useMediaQuery('(min-width:680px)');
+
+  const { courses } = useSavedCourses();
+
+  const inSession_savedCourses = courses
+    .filter((course) => course.term === term)
+    .filter((course) => course.lecture || course.lab || course.tutorial);
+
+  return (
+    <>
+      <Button
+        isDisabled={inSession_savedCourses.length > 0 ? false : true}
+        size="sm"
+        bg="blue.400"
+        color="white"
```

Comment:
```suggestion
       colorScheme="blue"
```
will handle dark mode